cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -2317,6 +2365,14 @@ const FunctionDoc utf8_lower_doc(
     "Transform input to lowercase",
     ("For each string in `strings`, return a lowercase version."), {"strings"});
 
+const FunctionDoc ascii_reverse_doc(
+    "Reverse ascii input",
+    ("For each ascii string in `strings`, return a reversed version."), {"strings"});
```

Comment:
For consistency, can we include a message similar to the other kernels about UTF-8 support? e.g.

```
     "This function assumes the input is fully ASCII.  It it may contain\n"
     "non-ASCII characters, use \"utf8_reverse\" instead."),
```